controllers/mtls_reconciler.go

```diff
@@ -0,0 +1,307 @@
+				},
+			},
+		}
+
+		unstructuredPeerAuth, err := controller.Destruct(peerAuth)
+		if err != nil {
+			logger.Error(err, "failed to destruct peer authentication", "status", "error")
+			return err
+		}
+		logger.Info("creating peer authentication resource", "status", "processing")
+		_, err = r.Client.Resource(istio.PeerAuthenticationResource).Namespace(kObj.Namespace).Create(eventCtx, unstructuredPeerAuth, metav1.CreateOptions{})
+		if err != nil {
+			if apiErrors.IsAlreadyExists(err) {
+				logger.Info("already created peer authentication resource", "status", "acceptable")
+			} else {
+				logger.Error(err, "failed to create peer authentication resource", "status", "error")
+				return err
+			}
+		}
+	}
+
+	// update the envoyfilters to either patch or remove based on previous logic
+	valueStruct, err := structpb.NewStruct(valueMap)
+	if err != nil {
+		logger.Info("problem processing patch for Envoy Filter")
+		return err
+	}
+	patch := &v1alpha32.EnvoyFilter_EnvoyConfigObjectPatch{
+		ApplyTo: v1alpha32.EnvoyFilter_CLUSTER,
+		Patch: &v1alpha32.EnvoyFilter_Patch{
+			Operation: operation,
+			Value:     valueStruct,
+		},
+	}
+
+	// add or remove the patch to each EnvoyFilter with prefix kuadrant- in the same namespace as your gateway.
+	for _, gateway := range gateways {
+
+		envoyFilters := lo.FilterMap(topology.Objects().Items(), func(item machinery.Object, _ int) (*v1alpha3.EnvoyFilter, bool) {
+			if rObj, isObj := item.(*controller.RuntimeObject); isObj {
+				if record, isRec := rObj.Object.(*v1alpha3.EnvoyFilter); isRec {
+					if strings.HasPrefix(record.Name, "kuadrant-") && gateway.GetNamespace() == record.Namespace {
+						return record, true
+					}
+				}
+			}
+			return nil, false
+		})
+
+		for _, envoyFilter := range envoyFilters {
+			envoyFilter.Spec.ConfigPatches = append(envoyFilter.Spec.ConfigPatches, patch)
+			unstructuredEnvoyFilter, err := controller.Destruct(envoyFilter)
+			if err != nil {
+				logger.Error(err, "failed to destruct limitador", "status", "error")
+				return err
+			}
+			logger.Info("creating limitador resource", "status", "processing")
+			_, err = r.Client.Resource(istio.EnvoyFiltersResource).Namespace(envoyFilter.Namespace).Update(eventCtx, unstructuredEnvoyFilter, metav1.UpdateOptions{})
+			if err != nil && !apiErrors.IsAlreadyExists(err) {
+				logger.Error(err, "failed to update envoyfilter resource", "status", "error")
+				return err
+			}
+		}
+
+	}
+	return nil
+}
+
+func (r *MTLSReconciler) deleteAllPeerAuthentications(ctx context.Context, peerAuthenticationObjs []machinery.Object, logger logr.Logger) {
+	for _, peerAuthentication := range peerAuthenticationObjs {
+		logger.V(1).Info(fmt.Sprintf("deleting peer authentication %s %s/%s", peerAuthentication.GroupVersionKind().Kind, peerAuthentication.GetNamespace(), peerAuthentication.GetName()))
+		mapping, err := r.restMapper.RESTMapping(peerAuthentication.GroupVersionKind().GroupKind())
```

Comment:
I don't think the restMapper is required. We know this information at compile time. A good example setting the up can be seen [here](https://github.com/Kuadrant/kuadrant-operator/blob/main/pkg/observability/schema.go)